### PR TITLE
Add optional/required column to AddFile and RemoveFile spec. 

### LINKED ISSUE #1566: [BUG] Protocol doesn't explicitly describe what fields are required or optional
## Bug

The [Add File and Remove File](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file) doesn't explicitly says if the fields are required or optional. Could we add a column in the table saying if a field is required or optional?

For example, size and modificationTime are required? 

### Willingness to contribute

The Delta Lake Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the Delta Lake code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the Delta Lake community.
- [X] No. I cannot contribute a bug fix at this time.

### PR DESCRIPTION
**Pull request by felipepessoto**
## Description

Resolves #1566

Add the optional/required column to AddFile and RemoveFile spec. I used the Scala source code as source of truth.
I also formatted the markdown table, so every column is aligned.

## How was this patch tested?

Docs only

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

**Comment by felipepessoto** [comment]
Regarding RemoveFile -> deletionTimestamp. It looks required and is present since the first commit of PROTOCOL.md, but in source code and recent protocol versions it is an Option[Long].

File: PROTOCOL.md [hunk 1]
```diff
@@ -332,16 +332,16 @@ For example, streaming queries that are tailing the transaction log can use this
...
-modificationTime | Long | The time this logical file was created, as milliseconds since the epoch
-dataChange | Boolean | When `false` the logical file must already be present in the table or the records in the added file must be contained in one or more `remove` actions in the same version
-stats | [Statistics Struct](#Per-file-Statistics) | Contains statistics (e.g., count, min/max values for columns) about the data in this logical file
-tags | Map[String, String] | Map containing metadata about this logical file
-deletionVector | [DeletionVectorDescriptor Struct](#Deletion-Vectors) | Either null (or absent in JSON) when no DV is associated with this data file, or a struct (described below) that contains necessary information about the DV that is part of this logical file.
+| Field Name       | Data Type                                            | Description                                                                                                                                                                                                                                                                            | optional/required |
```
Comment (scottsand-db) [inline PROTOCOL.md:335]: IMO all this whitespace just adds noise to this PR and makes it harder to review. We can add that in another PR .. or, at least another commit so we can still view the meat of this PR in a clean diff. But TBH, if we've avoided that convention (whitespace-padded-table) for years, I don't see why we would add it now. Please let me know if this is the latest markdown standard.

Comment (scottsand-db) [inline PROTOCOL.md:335]: `Optional/Required` (capitalized)

**Review by scottsand-db — COMMENTED** [review]
All of the required/optional entries LGTM.
Can you please revert the whitespace additions so this PR has less noise.

Thanks!

File: PROTOCOL.md (hunk 1 above)
Comment (felipepessoto) [reply PROTOCOL.md:335]: In https://github.com/delta-io/delta/blob/master/PROTOCOL.md#change-metadata we use lower-case, should I capitalize both?

Comment (scottsand-db) [reply PROTOCOL.md:335]: Cool. Then leave it lower-case. Thanks for checking!

Comment (felipepessoto) [reply PROTOCOL.md]: Whitespace reverted

**Comment by felipepessoto** [comment]
Done, thanks

**Review by vkorukanti — APPROVED** [review]
lgtm

**Pull request closed without merging**